spec/requests/user_data_spec.rb

```diff
@@ -157,4 +156,48 @@
     end
   end
 
+  describe 'request error messages' do
+    describe 'exception TokenNotValidError raised' do
```

Comment:
these inner `describe` blocks should be `context` blocks, as with each one you are setting up a *context* in which new conditions apply